src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +41,86 @@ public static void main(String[] args) throws IOException {
         LibSetup.setupLibraries();
         StartupParams params = new StartupParams(args);
 
+        checkUniqueInstance(params); //java 9
         checkJavaVersion(params);
         SwingUtilities.invokeLater(() -> new Main(params));
     }
 
+    private static void checkUniqueInstance(StartupParams params) throws IOException {
+        if (params.has(StartupParams.LaunchArg.NO_WARN)) return;
+
+        Path filePath = Paths.get("curr.pid");
+        long currentPid = ProcessHandle.current().pid();
+        ProcessHandle processHandle = ProcessHandle.current();
+        ProcessHandle.Info processInfo = processHandle.info();
+        long currentStartTime = processInfo.startInstant().map(Instant::toEpochMilli).orElse(0L);
+
+        if (Files.exists(filePath)) {
+            List<String> fileContent = Files.readAllLines(filePath, StandardCharsets.UTF_8);
+
+            if (!fileContent.isEmpty()) {
+                for (int i = 0; i < fileContent.size(); i++) {
+                    try {
+                        long filePid = 0;
+                        long fileStartTime = 0;
+                        String[] fileSplit = fileContent.get(i).split(" ");
+                        if (fileSplit.length == 2){
+                            filePid = Long.parseLong(fileSplit[0]);
+                            fileStartTime = Long.parseLong(fileSplit[1]);
+                        }
+
+                        ProcessHandle externalProcessHandle = ProcessHandle.of(filePid).orElse(null);
+                        long externalStartTime = 0;
+                        if (externalProcessHandle != null) {
+                            ProcessHandle.Info ExternalProcessInfo = externalProcessHandle.info();
+                            externalStartTime = ExternalProcessInfo.startInstant().map(Instant::toEpochMilli).orElse(0L);
+                        }
+
+                        if (ProcessHandle.of(filePid).isPresent() && externalStartTime == fileStartTime) {
+                            Files.writeString(filePath, "\n" + currentPid + " " + currentStartTime, StandardOpenOption.APPEND);
```

Comment:
just leave this one for the end, if you close the panel there's no need for the pid to be written at all